comp/core/workloadmeta/types.go

```diff
@@ -1027,6 +1027,13 @@ type EventBundle struct {
 	// Events gives the events in this bundle.
 	Events []Event
 
-	// Ch should be closed once the subscriber has handled the event.
+	// ch should be closed once the subscriber has handled the event.
 	Ch chan struct{}
 }
+
+// Acknowledge acknowledges that the subscriber has handled the event.
+func (e EventBundle) Acknowledge() {
+	if e.Ch != nil {
+		close(e.Ch)
+	}
```

Comment:
IMO you can still have a race between the `if` and the `close` if `Acknowledge()` is executed in 2 different goroutines.

A solution can be to add `sync.Once` in the EventBundle struct.

Reply:
Indeed but I think an `EventBundle` should be handled by a single goroutine.